libs/aws/langchain_aws/runnables/q_business.py

```diff
@@ -0,0 +1,160 @@
+    """Amazon Q client."""
+
+    application_id: str = None
+
+    _last_response: Dict = None  # Add this to store the full response
+    """Store the full response from Amazon Q."""
+
+    parent_message_id: Optional[str] = None
+
+    conversation_id: Optional[str] = None
+
+    chat_mode: str = "RETRIEVAL_MODE"
+
+    model_config = ConfigDict(
+        extra="forbid",
+    )
+
+    def __init__(
+        self,
+        region_name: Optional[str] = None,
+        credentials: Optional[Any] = None,
+        client: Optional[Any] = None,
+        application_id: str = None,
+        parent_message_id: Optional[str] = None,
+        conversation_id: Optional[str] = None,
+        chat_mode: str = "RETRIEVAL_MODE",
+    ):
+        self.region_name = region_name
+        self.credentials = credentials
+        self.client = client or self.validate_environment()
+        self.application_id = application_id
+        self.parent_message_id = parent_message_id
+        self.conversation_id = conversation_id
+        self.chat_mode = chat_mode
+
+    def invoke(
+        self,
+        input: Union[str,BaseMessage],
+        config: Optional[RunnableConfig] = None,
+        **kwargs: Any
+    ) -> BaseMessage:
+        """Call out to Amazon Q service.
+
+        Args:
+            input: The prompt to pass into the model.
+
+        Returns:
+            The string generated by the model.
+
+        Example:
+            .. code-block:: python
+
+            model = AmazonQ(
+                credentials=your_credentials,
+                application_id=your_app_id
+            )
+            response = model.invoke("Tell me a joke")
+        """
+        try:
+            # Prepare the request            
+            request = {
+                'applicationId': self.application_id,
+                'userMessage': self.convert_langchain_messages_to_q_input(input), # Langchain's input comes in the form of an array of "messages". We must convert to a single string for Amazon Q's use
+                'chatMode': self.chat_mode,
+            }
+            if self.conversation_id:
+                request.update({
+                    'conversationId': self.conversation_id,
+                    'parentMessageId': self.parent_message_id,
+                })
+
+            # Call Amazon Q
+            response = self.client.chat_sync(**request)
+            self._last_response = response
+
+            # Extract the response text
+            if 'systemMessage' in response:
+                return response["systemMessage"]
+            else:
+                raise ValueError("Unexpected response format from Amazon Q")
+
+        except Exception as e:
+            if "Prompt Length" in str(e):
+                logging.info(f"Prompt Length: {len(input)}")
+                logging.info(f"""Prompt:
+                {input}""")
+            raise ValueError(f"Error raised by Amazon Q service: {e}")
+
+    def get_last_response(self) -> Dict:
```

Comment:
Not clear how `_last_response` is useful. Also, should this be a property?

```suggestion
    @property
    def last_response(self) -> Dict:
```

Reply:
It is used to retrieve the reference the response details besides the message. For example, a developer may want to re-use the conversationId or pass the parentMessageId to the next request to keep the same conversation going. So they would need to get this information. ([link](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/qbusiness/client/chat_sync.html#:~:text=parentMessageId%20(string)%20%E2%80%93%20The%20identifier%20of%20the%20previous%20system%20message%20in%20a%20conversation.))

We want it to be easy for developers to invoke the runnable and return a string response containing the AI message. 
But if they have an edge use case where they need other information this is stored so they can retrieve that as well.

Reply:
See my comment about embedding this in AIMessage.

Reply:
implemented